### ISSUE TITLE
Aspects entry sometimes edits wrong Aspect field.

### ISSUE THREAD

**Issue by johnhelmuth** [Earlo/fate#11]
Aspects entry sometimes edits wrong Aspect field.

### Reproduce
* Create a new character
* Click into the High Concept field
* Hit tab twice to navigate to the first Additional Aspect field
* Type your aspect

### Expected
The aspect typed should be in the 3rd field with the hint "Additional Aspect".

### Actual
The first few characters typed show up in the first and second Aspect fields ("High Concept" and "Trouble".)

### Additional notes
This is reproducible on commits `2abc195` through `46e3c0a` of the `main` branch. (I didn't dig back further.

If you paste instead of type the Aspect, the whole thing shows up in the first Aspect field.

Also reproducible on the Campaign Aspect fields.

Tested on Safari (16.6), Chrome (117.0.5938.88), and Firefox (109.01 and 117.0.1) on Mac OS X Ventura (13.5.2) on an M1 processor.

**Comment by Earlo** [comment]
Confirmed that this happens. Really weird behaviour.

**Comment by johnhelmuth** [comment]
Dug into it a little bit.  I think it's because the `onChange` propagation from the `SoloInput` => `Input` =>  `AspectInput.handleAspects()` isn't tracking the `index` value correctly.

I'll take this one on, since I've already analyzed it so far.

**Comment by johnhelmuth** [comment]
Actually because the empty array of aspects is trying to insert at a specific index with the `Array.slice()` method:

```javascript
[
      ...aspects.slice(0, index),
      value,
      ...aspects.slice(index + 1),
    ]
```

When `aspects` is an empty array, it puts the first `onChange()` value into the first element, since `aspects.slice(0,4)` is an empty array itself.

**Comment by Earlo** [comment]
Fixed this in #11
